ctapipe/reco/impact_distance.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+import numpy as np
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+
+from ..containers import ReconstructedGeometryContainer
+from ..coordinates import GroundFrame, TiltedGroundFrame
+from ..instrument.subarray import SubarrayDescription
+
+__all__ = ["shower_impact_distance"]
+
+
+def impact_distance(point: np.ndarray, direction: np.ndarray, test_points: np.ndarray):
+    """Compute impact distance from a line defined by a point and a direction vector
+    with an array of points
+
+    Parameters
+    ----------
+    point: np.ndarray
+       point defining the line
+    direction: np.ndarray
+       direction vector
+    test_points: np.ndarray
+       array of test points
+    """
+    return np.linalg.norm(
+        np.cross(test_points - point, direction), axis=1
+    ) / np.linalg.norm(direction)
+
+
+def shower_impact_distance(
+    shower_geom: ReconstructedGeometryContainer, subarray: SubarrayDescription
+):
+    """computes the impact distance of the shower axis to the telescope positions
+
+    Parameters
+    ----------
+    shower_geom: ReconstructedGeometryContainer
+        reconstructed shower geometry, must contain a core and alt/az position
+    subarray: SubarrayDescription
+        SubarrayDescription from which to extract the telescope positions
+
+    Returns
+    -------
+    np.ndarray:
+       array of impact distances to each telescope (packed by tel index)
+    """
+
+    # core position in the ground frame
+    core_position = np.array(
+        [
+            shower_geom.core_x.to_value(u.m),
+            shower_geom.core_y.to_value(u.m),
+            0,
+        ]
+    )
+
+    # sky direction of the shower (defines the shower axis)
+    sky_direction = SkyCoord(
+        alt=shower_geom.alt, az=shower_geom.az, frame="altaz"
```

Comment:
Hi @kosack 
I think I know where the problem might be. 
Geographic azimuth is computed from the north axis towards east, so e.g. alt=0, az=45 deg should correspond to a vector [0.707, -0.707, 0], but when you do SkyCoords (alt, az, frame ="altaz").cartesian.xyz it makes [0.707, 0.707, 0], like it would be computed from x direction towards y. 
If I add here a minus sign before shower_geom.az the results of both methods agree (and I hope that provide correct values :-) )

Reply:
Maybe related:

Reply:
@jsitarek But we don't use `AltAz.cartesian` in the TiltedFrame transformations, we build a rotation matrix directly from `alt` and `az`...

Reply:
I do use the cartesian attribute when turning the alt/az to a cartesian direction. I should probably just to the matrix explicitly to avoid SkyCoord slowness

Reply:
https://github.com/astropy/astropy/issues/5300

Reply:
Ok should agree now. And the 3d method is even faster without the SkyCoord overhead.